src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -1830,6 +1836,32 @@ namespace QgsWms
       return false;
     }
 
+    // Sanity check from internal QImage checks (see qimage.cpp)
+    // this is to report a meaningful error message in case of
+    // image creation failure and to differentiate it from out
+    // of memory conditions.
+
+    // depth for now it cannot be anything other than 32, but I don't like
+    // to hardcode it: I hope we will support other depths in the future.
+    uint depth = 32;
+    switch ( mWmsParameters.format() )
+    {
+      case QgsWmsParameters::Format::JPG:
+      case QgsWmsParameters::Format::PNG:
+      default:
+        depth = 32;
+    }
+
+    const int bytes_per_line = ( ( width * depth + 31 ) >> 5 ) << 2; // bytes per scanline (must be multiple of 4)
+
+    if ( INT_MAX / depth < width
+         || bytes_per_line <= 0
+         || height <= 0
+         || INT_MAX / uint( bytes_per_line ) < height
+         || INT_MAX / sizeof( uchar * ) < uint( height ) )
```

Comment:
``` c++
  || INT_MAX / sizeof( uchar * ) < uint( height ) )
```

Is it wanted to use the size of the pointer instead of using the size of `uchar` type itself?

Reply:
@pblottiere That's a (good) question for Qt core developers: as I mentioned in the commit that code is taken verbatim from the check that `QImage` 5.9.3 does internally.

The idea is that if that check fails we get a null image, so, instead of failing later when we cannot tell wether it was an out-of-memory or a size failure, we do that check inside our size check function so that we can report to the user a meaningful message that the the size was wrong.

So, the check needs to be the same that is done in Qt core, that's why I took it verbatim from there.

Btw, I see that in 5.10 is is apparently changed slightly https://github.com/qt/qtbase/blob/5.10/src/gui/image/qimage.cpp#L130
This is 5.9:
https://github.com/qt/qtbase/blob/5.9/src/gui/image/qimage.cpp#L130

What do you think we should do?

Reply:
> What do you think we should do?

I think that `std::numeric_limits<int>::max()` should be prefered over `INT_MAX`.

However, I don't have a clue about `std::numeric_limits<qsizetype>::max()`...